OpenRA.Mods.Common/Traits/Buildings/RallyPoint.cs

```diff
@@ -84,7 +93,13 @@ public RallyPoint(Actor self, RallyPointInfo info)
 
 		void INotifyCreated.Created(Actor self)
 		{
-			effect = new RallyPointIndicator(self, this);
+			effect = CreateRallyPointIndicator(self);
+			validator = self.TraitOrDefault<IRallyPointValidator>();
```

Comment:
this feels wrong. This should really be the job of either this trait or its parents. Not 1 outside info trait

Reply:
this implementation also doesn't feel very extendable. Shouldn't the targeter be overridable as well?

Reply:
As for the rally point location validation, you're right, it actually doesn't need custom interface, just making `Orders` property virtual is enough. But in order to make bots working with this, the validator is necessary, so the validation can occur in `BaseBuilderBotModule` too.

> this feels wrong. This should really be the job of either this trait or its parents. Not 1 outside info trait

What do you mean by this? Are you referring to the validator or `CreateRallyPointIndicator()` method?

Reply:
I'm referring to IRallyPointValidator